pkg/ruler/scheduler.go

```diff
@@ -206,7 +209,16 @@ func (s *scheduler) addNewConfigs(now time.Time, cfgs map[string]configs.Version
 	hasher := fnv.New64a()
 
 	for userID, config := range cfgs {
-		rulesByFilename, err := config.Config.Parse()
+		rulesByGroup := map[string][]rules.Rule{}
+		var err error
+		switch s.ruleFormatVersion {
```

Comment:
Maybe factor this switch out to a function in `config`, so we don't have to edit this file when v3 is added?
(And it would help keep down the complexity of this function)

Reply:
Great idea, I factored that into a `Parse()` method in `config`, which also got rid of those `switch`es in other places.